### PR TITLE
AvaloniaPropertyChangedEventArgs is now a readonly struct

### LINKED ISSUE #3169: AvaloniaPropertyChangedEventArgs can be a struct
Like its corresponding WPF type, AvaloniaPropertyChangedEventArgs can be a struct to reduce memory allocations.
Avalonia runs fine with this change.

### PR DESCRIPTION
**Pull request by mstr2**
## Fixed issues
Fixes #3169

### CODE REVIEW

**Comment by grokys** [comment]
Note that this is a breaking change as `AvaloniaPropertyChangedEventArgs` no longer derives from `EventArgs`.

File: src/Avalonia.Base/AvaloniaObject.cs [hunk 1]
```diff
@@ -759,8 +759,6 @@ private AvaloniaProperty GetRegistered(AvaloniaProperty property)
         /// </remarks>
         private void ParentPropertyChanged(object sender, AvaloniaPropertyChangedEventArgs e)
```
Comment (kekekeks) [inline src/Avalonia.Base/AvaloniaObject.cs]: It should be used with `in` modifier everywhere now to avoid making copies

**Comment by mstr2** [comment]
I've benchmarked both the original `class` version as well as the version with `readonly struct` + `in`. While the benchmark using the `class` version has twice as many Gen0 collections and twice as much allocated memory, it's actually ever so slightly faster than the `struct` version. Probably not enough of a reason to change the API.

**Comment by grokys** [comment]
I'm currently working on a refactor of the binding system to make it typed. For this refactor, I'm going to need to be able to inherit from `AvaloniaPropertyChangedEventArgs` so a struct won't work there. I'm going to close this PR, we can reopen if we decide that we do need to it be a struct after all.

Thanks for the PR anyway, @mstr2 !

**Pull request closed without merging**